project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestExportModules.java

```diff
@@ -0,0 +1,123 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.projects.test;
+
+import io.ballerina.projects.PackageCompilation;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.ProjectEnvironmentBuilder;
+import io.ballerina.projects.bala.BalaProject;
+import io.ballerina.projects.directory.BuildProject;
+import io.ballerina.projects.repos.TempDirCompilationCache;
+import org.ballerinalang.test.BCompileUtil;
+import org.ballerinalang.test.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+/**
+ * Contains cases to test export modules.
+ *
+ * @since 2.0.0
+ */
+public class TestExportModules {
+    private static final Path RESOURCE_DIRECTORY = Paths.get("src", "test", "resources");
+    private static final Path EXPORT_MODULES = RESOURCE_DIRECTORY.resolve("export_modules");
+
+    @Test(description = "tests loading a valid bala project with export in package.json")
+    public void testBalaProjectWithExportModules() {
+        Path balaPath = EXPORT_MODULES.resolve("bala_with_export").resolve("foo-winery-any-0.1.0.bala");
```

Comment:
Since we now work with the extracted bala, shall we use the same for tests as well?.

Reply:
Added